SubmitUserActivites.cs

```diff
@@ -0,0 +1,342 @@
+        }
+        public static void InputUtilities(int ID, List<dynamic> utilities, MySqlConnection connection, string date, string query) {
+            // new command
+            using MySqlCommand command = new(query, connection);
+            // varaibles that dont change for each unique foods
+            int userID = ID;
+            const string activityType = "utility";
+            Functions calculations = new();
+
+            foreach (var utility in utilities)
+            {
+                // get the utility type
+                string method = utility.UtilityType;
+                // get hours used
+                string amount = utility.Hours;
+                // calculate the utilities emission
+                double emission = calculations.UtilitiesCalculation(method, amount);
+                if (emission == -1)
+                {
+                    break;
+                }
+
+                // insert info in Activities Table
+                command.CommandText = query;
+                command.Parameters.AddWithValue("@ID", ID);
+                command.Parameters.AddWithValue("@UserID", userID);
+                command.Parameters.AddWithValue("@ActivityType", activityType);
+                command.Parameters.AddWithValue("@Method", method);
+                command.Parameters.AddWithValue("@Amount", amount);
+                command.Parameters.AddWithValue("@DateTime", date);
+                command.Parameters.AddWithValue("@Emission", emission);
+                command.ExecuteNonQuery();
+            }
+        }
+        private static void UpdateDailyEmissions(int ID, MySqlConnection connection, string date)
+        {
+            // get the total Emissions from the day
+            double addedTotalEmissions = GetTotalEmissions(ID, connection, date);
+            UpdateOrSetDailyEmissions(ID, connection, date, addedTotalEmissions);
+        }
+        private static double GetTotalEmissions(int ID, MySqlConnection connection, string date)
+        {
+            // new command
+            string query = "SELECT Emission FROM Activities WHERE UserID = '" + ID + "' AND DateTime = '" + date + "'";
+            using MySqlCommand command = new(query, connection);
+
+            double totalAddedEmissions = 0;
+            // set command and reader at the Users Table
+            command.CommandText = query;
+            using MySqlDataReader reader = command.ExecuteReader();
+            while (reader.Read())
+            {
+                double currEmission = reader.GetDouble("Emission");
+                totalAddedEmissions += currEmission;
+            }
+            return totalAddedEmissions;
+        }
+         private static void UpdateOrSetDailyEmissions(int ID, MySqlConnection connection, string date, double addedTotalEmissions)
+        {
+            // read table DailyEmissions
+            string query = "SELECT * FROM DailyEmissions WHERE UserID = '" + ID + "' AND DateTime = '" + date + "'";

```

Comment:
Check the SELECT FOR UPDATE keyword for MySQL. This prevent race condition in sql database.